### PR TITLE
[AWS] Update AWS cloud.region parsing

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

The AWS-S3 input sets the `cloud.region` field to the region the S3 bucket is located. This is not always the same as the region the event originated per https://discuss.elastic.co/t/filebeat-aws-cloudtrail-processor-parses-incorrect-aws-region-from-logs/312150.  The current ingest pipelines only use a rename processor which doesn't work if the field already exists.  This changes the processor a set to override from the input. Also adds additional parsing for datastreams that don't already have `cloud.region` set.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by sypste** [comment]
Relates to https://github.com/elastic/beats/issues/32609

**Review by efd6 — APPROVED** [review]

**Review by kaiyan-sheng — APPROVED** [review]

File: packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -41,6 +41,12 @@ processors:
         S3KEY: "[a-zA-Z0-9\\/_\\.\\-%+]+"
         S3ID: "[a-zA-Z0-9\\/_\\.\\-%+=]+"
         S3VERSION: "[a-zA-Z0-9.]+"
+  - grok:
+      field: aws.s3access.host_header
+      ignore_missing: true
+      ignore_failure: true
+      patterns:
+        - ^%{DATA}s3\.%{DATA:cloud.region}\.%{DATA}$
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:49]: @legoguy1000 Should we add a check to see if aws.s3acccess.host_header exists?

Comment (legoguy1000) [reply packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:49]: Isnt that the idea of the ignore_missing??

**Pull request merged**